### PR TITLE
update enterpriseTests setup

### PR DESCRIPTION
**Pull request by wfurt**
this change should have no functional impact but it is stepping stone for some future improvements. 

There are essentially 2 changes here: 
1. Currently the web server expect Kerberos auth for everything. This change moves that requirement to /auth/kerberos. This will allow us to have additional authentication setups under /auth/ as well is opens possibly to test other features - like compression - easier.

2. this is somewhat internal optimization. httpd:2.4 requires Debian while the other containers use Ubuntu 18.04. This change unifies everything on Ubuntu18 so we do not need to fetch two base OS images for each run. 

Most of the changes in apache2.conf  are to deal with damages from the Debian->Ubuntu migration and to support goal above. 

contributes to #47351
contributes to #29270
contributes to #17845
contributes to #31133

### CODE REVIEW

**Comment by wfurt** [comment]
Since nobody got this this yet, I decide to put it more changes toward the end goal and I added end2end test for NTLM and Digest auth. 

The setup now creates PDC on Samba and use win bind to do the authentication. Unfortunately the win bind model does not play nicely with the Kerberos so I end up splitting them to two operate processes. Also the old 'password' did not meet complexity criteria for PDC so I change that for all methods.

**Comment by karelz** [comment]
@antonfirsov can you please help code review? Thanks!

File: src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+if [ "$1" == "-debug" ]; then
+  while [ 1 ];do
+    sleep 10000
+  done
+fi
+
+if [ "$1" == "-DNTLM" ]; then
+  ./setup-pdc.sh
+  /usr/sbin/apache2 -DALTPORT "$@"
+  shift
```
Comment (antonfirsov) [inline src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh:16]: Do I get it right that the goal here is to start two apache instances? (This one in the background.)

Can you add an explanation why are we doing so?

**Review by antonfirsov — COMMENTED** [review]
I think it would be nice to add some comments, or maybe a short update in `README.md` regarding the new setup, otherwise looks good.

File: src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+echo "$@" > /tmp/args
 
 cp /SHARED/apacheweb.keytab /etc/krb5.keytab
 
-exec httpd -DFOREGROUND "$@"
+if [ "$1" == "-debug" ]; then
+  while [ 1 ];do
+    sleep 10000
+  done
+fi
```
[the remark this answers is not in the document]
Comment (wfurt) [reply src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh:11]: this is not used in production. This is debug hook when when used locally. 
I probably should update README tor remove it.

File: src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh (hunk 1 above)
Comment (wfurt) [reply src/libraries/Common/tests/System/Net/EnterpriseTests/setup/apacheweb/run.sh:16]: Yes. we did run two instances before but the mechanic was different. This comes from the fact that we use single docker image to run standard and non-default service.

**Review by antonfirsov — APPROVED** [review]
LGTM, rerunning CI to make sure to exclude infra changes breaking the PR.

**Pull request merged**